torch/csrc/distributed/autograd/context/dist_autograd_context.h

```diff
@@ -40,9 +42,23 @@ class TORCH_API DistAutogradContext {
   DistAutogradContext(DistAutogradContext&&) = delete;
   DistAutogradContext& operator=(DistAutogradContext&&) = delete;
 
+  static_assert(
+      std::numeric_limits<rpc::worker_id_t>::max() <=
+          std::numeric_limits<int16_t>::max(),
+      "Range of possible workerIDs exceeds int16_t limit. This could cause "
+      "memory issues at runtime. Consider changing known_worker_set_type from "
+      "std::bitset to std::unordered_map");
+  using known_worker_set_type =
```

Comment:
Let's keep this simple and just use `std::unordered_set<rpc::worker_id_t>` here. Also, I feel `WorkerSet` is a much better (and shorter) name for the type than `known_worker_set_type`.